File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack();
        for (String token : tokens) {
            if (token.equals("+")) {
                stack.push(stack.pop() + stack.pop());
            } else if (token.equals("-")) {
                Integer a = stack.pop();
                Integer b = stack.pop();
                stack.push(b-a);
            } else if (token.equals("*")) {
                stack.push(stack.pop() * stack.pop());
            } else if (token.equals("/")) {
                Integer a = stack.pop();
                Integer b = stack.pop();
                stack.push(b / a);
            } else {
                stack.push(Integer.valueOf(token));
            }
        }
        return stack.pop().intValue();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.kt
class Solution {
    fun evalRPN(tokens: Array<String>): Int {
        val stack = Stack<Int>()
        for (token in tokens) {
            if (token == "+") {
                stack.push(stack.pop() + stack.pop())
            } else if (token == "*") {
                stack.push(stack.pop() * stack.pop())
            } else if (token == "-") {
                val val1 = stack.pop();
                val val2 = stack.pop();
                stack.push(val2 - val1)
            }  else if (token == "/") {
                val val1 = stack.pop();
                val val2 = stack.pop();
                stack.push(val2 / val1)
            } else {
                stack.push(token.toInt())
            }
        }
        return stack.pop();
    }
}